cmd/serve.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/mount"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/go-connections/nat"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+const (
+	ImagePath        = "tigrisdata/tigrisdb"
+	FDBImagePath     = "foundationdb/foundationdb:6.3.23"
+	volumeName       = "fdbdata"
+	FDBContainerName = "tigrisdb-cli-fdb"
+	ContainerName    = "tigrisdb-cli-server"
+)
+
+var ImageTag = "1.0.0-alpha.4"
+
+func ensureVolume(cli *client.Client) {
+	ctx := context.Background()
+
+	volumes, err := cli.VolumeList(ctx, filters.NewArgs())
+	if err != nil {
+		log.Fatal().Err(err).Msg("error listing volumes")
+	}
+
+	found := false
+	for _, v := range volumes.Volumes {
+		if v.Name == volumeName {
+			found = true
+		}
+	}
+
+	if !found {
+		_, err := cli.VolumeCreate(ctx, volumetypes.VolumeCreateBody{
+			Driver: "local",
+			Name:   volumeName,
+		})
+		if err != nil {
+			log.Fatal().Err(err).Msg("error creating docker volume")
+		}
+	}
+}
+
+func stopContainer(client *client.Client, cname string) {
+	ctx := context.Background()
+
+	if err := client.ContainerStop(ctx, cname, nil); err != nil {
+		if !errdefs.IsNotFound(err) {
+			log.Fatal().Err(err).Str("name", cname).Msg("error stopping container")
+		}
+	}
+
+	opts := types.ContainerRemoveOptions{
+		RemoveVolumes: true,
+		Force:         true,
+	}
+
+	if err := client.ContainerRemove(ctx, cname, opts); err != nil {
+		if !errdefs.IsNotFound(err) {
+			log.Fatal().Err(err).Str("name", cname).Msg("error stopping container")
+		}
+	}
+}
+
+func startContainer(cli *client.Client, cname string, image string, volumeMount string, port string) string {
+	ctx := context.Background()
+
+	reader, err := cli.ImagePull(ctx, image, types.ImagePullOptions{})
+	if err != nil {
+		log.Fatal().Err(err).Str("image", image).Msg("error pulling docker image")
+	}
+	defer func() { _ = reader.Close() }()
+
+	_, _ = io.Copy(os.Stdout, reader)
+
+	m := mount.Mount{
+		Type:   mount.TypeVolume,
+		Source: volumeName,
+		Target: volumeMount,
+	}
+
+	p, err := nat.ParsePortSpec(port)
+	if err != nil {
+		log.Fatal().Err(err).Str("image", image).Msg("error parsing port")
+	}
+
+	pm := nat.PortMap{}
+
+	for _, v := range p {
+		pm[v.Port] = []nat.PortBinding{v.Binding}
+	}
+
+	resp, err := cli.ContainerCreate(ctx,
+		&container.Config{
+			Image: image,
+			Tty:   false,
+		},
+		&container.HostConfig{
+			Mounts:       []mount.Mount{m},
+			PortBindings: pm,
+		}, nil, nil, cname)
+	if err != nil {
+		log.Fatal().Err(err).Str("image", image).Msg("error creating container docker image")
+	}
+
+	if err = cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+		log.Fatal().Err(err).Str("image", image).Msg("error starting docker image")
+	}
+
+	return resp.ID
+}
+
+var serveCmd = &cobra.Command{
+	Use:   "serve [port] [tag]",
```

Comment:
What do you think about having a different command? Something that is clear from the command as well that it will bring up local server?

Reply:
Also recommend changing `tag` to `version` as that will be more clearer.

Reply:
Something like:

`tigrisdb-cli run-local` or
`tigrisdb-cli start-local`

Reply:
```
tigrisdb-cli server up
tigrisdb-cli server down
```

```
tigrisdb-cli local up
tigrisdb-cli local down
```

?

Reply:
```
Also recommend changing tag to version as that will be more clearer.
```
This is now docker tag (1.0.0-alpha.2) and it's different from version (v1.0.0-alpha.2)
if we want it to be version we would need to convert it (cut v from the beginning).

Reply:
`tigrisdb-cli local up` looks good to me

Reply:
Since tag is always generated from the version, what do you think is better? Accepting the version with `v` and cutting it? Or leave it as tag?

Reply:
Updated the diff, now it accepts both tag and version.